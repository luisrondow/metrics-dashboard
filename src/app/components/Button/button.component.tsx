import { Button as StyledButton } from './button.styles';

type ButtonProps = {
  children: React.ReactNode;
  onClick: () => void;
};

const Button = ({ children, onClick }: ButtonProps) => {
  return <StyledButton onClick={onClick}>{children}</StyledButton>;
};

export default Button;
