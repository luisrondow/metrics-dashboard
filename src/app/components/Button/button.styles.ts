import styled from "styled-components";

export const Button = styled.button`
  background-color: var(--primary-color);
  border: none;
  color: #FFF;
  padding: var(--space-sm) var(--space-md);
  text-align: center;
  text-decoration: none;
  display: inline-block;
  font-size: 1rem;
  cursor: pointer;
  border-radius: 4px;
  &:hover {
    background-color: var(--primary-color-light);
  }
`;