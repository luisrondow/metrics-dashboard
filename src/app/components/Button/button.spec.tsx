import { render, screen, fireEvent } from '@testing-library/react';
import Button from './button.component';

describe('Button Component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(<Button onClick={() => {}}>Click me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders button with children', () => {
    render(<Button onClick={() => {}}>Click me</Button>);
    const buttonElement = screen.getByText('Click me');
    expect(buttonElement).toBeInTheDocument();
  });

  it('calls onClick when button is clicked', () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>Click me</Button>);
    const buttonElement = screen.getByText('Click me');
    fireEvent.click(buttonElement);
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
