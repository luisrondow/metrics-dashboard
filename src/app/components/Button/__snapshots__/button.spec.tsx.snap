// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-e7b71969-0 oTGwr"
  >
    Click me
  </button>
</DocumentFragment>
`;
